internal/worker/client.go

```diff
@@ -222,13 +226,8 @@ func (c *Client) workerHeartbeat() {
 			}
 		}
 
-		url, err := c.serverURL.Parse(fmt.Sprintf("workers/%s/status", workerID()))
-		if err != nil {
-			logrus.Errorf("Error parsing worker status: %v", err)
```

Comment:
Even if we close this PR I think we should still fix this log message, because I think it's quite misleading. Unless I miss something here this can only happen if workerUUID() return something that creates an error from an URL parse - which should never happen but even if it does the error should be something like: `cannot construct URL for worker status` or similar and not be about parsing the worker status. When I read this I assumed `c.server.Parse()` would actually reach out to the worker which it does not (which partly promoted me to this flurry of PRs that (hopefully) make the code more explicit.